src/main/java/entities/Player.java

```diff
@@ -85,12 +86,30 @@ public Card playCard(int id) {
         return null;
     }
 
+    public boolean spendEssence(int num) {
+        if (num > getNumOfEssence()) {
+            return false;
+        }
+
+        Iterator<Card> iterator = hand.iterator();
+        while (iterator.hasNext() && num > 0) {
+            Card card = iterator.next();
+            if (card instanceof EssenceCard) {
+                iterator.remove();
+                num--;
+            }
+        }
+
+        return true;
+    }
+
     public void setStructure(StructureCard structure) {
         this.structure = structure;
     }
 
     public StructureCard getStructure() {
         return structure;
+
```

Comment:
You might wanna remove this line, this is probably by mistake but it's just a little inconsistent with the style.
```suggestion
```